### PR TITLE
ffmpeg: add `mingw` support 

### PR DESCRIPTION
**Pull request by seanybaggins**
## Description of changes
Add mingw support for ffmpeg.
Part of a larger effort to add mingw support for qtmultimedia
Relevant for https://github.com/NixOS/nixpkgs/issues/274274

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.ffmpeg

**Comment by seanybaggins** [comment]
@wegank @Artturin @bjornfor

**Review by bjornfor — APPROVED** [review]
LGTM.

Nit:
* Commit messages should be imperative style: "ran nixpkgs-fmt" -> "run nixpkgs-fmt".
* Commit message bodies should be full sentences: please end with a dot.

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 1]
```diff
@@ -82,13 +90,13 @@
 , withSrt ? withHeadlessDeps # Secure Reliable Transport (SRT) protocol
 , withSsh ? withHeadlessDeps # SFTP protocol
 , withSvg ? withFullDeps # SVG protocol
-, withSvtav1 ? withHeadlessDeps && !stdenv.isAarch64 # AV1 encoder/decoder (focused on speed and correctness)
+, withSvtav1 ? withHeadlessDeps && !stdenv.isAarch64 && !stdenv.hostPlatform.isMinGW # AV1 encoder/decoder (focused on speed and correctness)
 , withTensorflow ? false # Tensorflow dnn backend support
 , withTheora ? withHeadlessDeps # Theora encoder
-, withV4l2 ? withHeadlessDeps && !stdenv.isDarwin # Video 4 Linux support
+, withV4l2 ? withHeadlessDeps && !stdenv.isDarwin && !stdenv.hostPlatform.isMinGW # Video 4 Linux support
```
Comment (wegank) [inline pkgs/development/libraries/ffmpeg/generic.nix:96]:
```suggestion
, withV4l2 ? withHeadlessDeps && stdenv.isLinux # Video 4 Linux support
```

**Review by jopejoe1 — CHANGES_REQUESTED** [review]
A few things:

- Conflicting with changes in staging (due to formating)
- the formater run does not follow the new formating rules defiend in https://github.com/NixOS/rfcs/pull/166

The mingw changes them self look fine.

**Comment by seanybaggins** [comment]
@jopejoe1 should I revert the formatting?

**Comment by jopejoe1** [comment]
Yes please, the formating should be done when both branches are in sync.

**Comment by Atemu** [comment]
Please revert the formatting either way. We will soon have a mass format of all code using the new standardised RFC166 style anyways.

**Comment by seanybaggins** [comment]
> Does ffmpeg build and run successfully on/for Windows with this btw?

I have not directly tested ffmpeg, but I have tested a qt application that uses ffmpeg as one of its dependencies. See 
https://github.com/seanybaggins/qt-package-overalys

**Comment by seanybaggins** [comment]
> Please revert the formatting either way. We will soon have a mass format of all code using the new standardised RFC166 style anyways.

Done

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.ffmpeg

**Comment by wegank** [comment]
> We will soon have a mass format of all code using the new standardised RFC166 style anyways.

Nice, but I hope the PR won't be stalled like the one for RFC 140 was...

**Comment by seanybaggins** [comment]
@Artturin I think there is a bug on how dependencies get copied with mingwW64 builds.
```
nix build "github:nixos/nixpkgs/master#pkgs.pkgsCross.mingw32.xvidcore"

ls result/lib/
libgcc_s_sjlj-1.dll  libmcfgthread-1.dll  xvidcore.dll
```

```
nix build "github:nixos/nixpkgs/master#pkgs.pkgsCross.mingwW64.xvidcore"

ls result/lib/
xvidcore.dll
```

I believe the missing dlls is what is causing ffmpeg to fail for mingwW64 but succeed for mingw32.

**Comment by Atemu** [comment]
> I believe the missing dlls is what is causing ffmpeg to fail for mingwW64 but succeed for mingw32.

Could you set meta.broken accordingly?

**Comment by seanybaggins** [comment]
> > I believe the missing dlls is what is causing ffmpeg to fail for mingwW64 but succeed for mingw32.
> 
> Could you set meta.broken accordingly?

Done

**Comment by jopejoe1** [comment]
You re-added the formatting with your rebase

**Comment by seanybaggins** [comment]
> You re-added the formatting with your rebase

Yeah. My bad. Editor administration fail. Fixed.

**Review by jopejoe1 — APPROVED** [review]
Looks good to me now

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 2]
```diff
@@ -71,7 +71,7 @@
 , withOpenmpt ? withFullDeps # Tracked music files decoder
 , withOpus ? withHeadlessDeps # Opus de/encoder
 , withPlacebo ? withFullDeps && !stdenv.isDarwin # libplacebo video processing library
-, withPulse ? withSmallDeps && !stdenv.isDarwin # Pulseaudio input support
+, withPulse ? withSmallDeps && !stdenv.isDarwin && !stdenv.hostPlatform.isMinGW # Pulseaudio input support
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ffmpeg/generic.nix:74]:
```suggestion
, withPulse ? withSmallDeps && stdenv.isLinux # Pulseaudio input support
```

Isn't pulse more a linux thing?

Comment (jopejoe1) [reply pkgs/development/libraries/ffmpeg/generic.nix:74]: Pulse is mainly developed for linux but has support for others like FreeBSD and even MacOS and Windows.

**Comment by seanybaggins** [comment]
> The resulting binary does not appear to work in WINE:
> Log
> 
> ```
> 002c:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
> 002c:fixme:winediag:loader_init wine-staging 9.4 is a testing version containing experimental patches.
> 002c:fixme:winediag:loader_init Please mention your exact version when filing bug reports on winehq.org.
> 004c:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
> 0054:fixme:actctx:parse_depend_manifests Could not find dependent assembly L"Microsoft.Windows.Common-Controls" (6.0.0.0)
> 004c:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
> 004c:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
> 004c:err:ole:apartment_get_local_server_stream Failed: 0x80004002
> 004c:err:ole:start_rpcss Failed to open RpcSs service
> 0054:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hr 0x80004002
> 0054:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, hr 0x80004002
> 0054:err:ole:apartment_get_local_server_stream Failed: 0x80004002
> 007c:err:ntoskrnl:ServiceMain Failed to load L"C:\\windows\\system32\\win32k.sys"
> 007c:err:ntoskrnl:ServiceMain Failed to load L"C:\\windows\\system32\\drivers\\dxgkrnl.sys"
> 007c:err:ntoskrnl:ServiceMain Failed to load L"C:\\windows\\system32\\drivers\\dxgmms1.sys"
> 00e4:fixme:msg:pack_message msg 14 (WM_ERASEBKGND) not supported yet
> wine: configuration in L"/tmp/wine" has been updated.
> 0134:fixme:wineusb:add_usb_device Interface 2 has 2 alternate settings; using the first one.
> 0134:fixme:wineusb:add_usb_device Interface 0 has 2 alternate settings; using the first one.
> 0134:fixme:wineusb:add_usb_device Interface 0 has 2 alternate settings; using the first one.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 0024:err:module:import_dll Library libglib-2.0-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libharfbuzz-0.dll") not found
> 0024:err:module:import_dll Library libharfbuzz-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 010c:fixme:wineusb:query_id Unhandled ID query type 0x5.
> 0024:err:module:import_dll Library libexpat-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libfontconfig-1.dll") not found
> 0024:err:module:import_dll Library libfontconfig-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libssh.dll") not found
> 0024:err:module:import_dll Library libssh.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libwinpthread-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libsrt.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libp11-kit-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libidn2-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libunistring-5.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libtasn1-6.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libintl-8.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libgnutls-30.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library avformat-60.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 0024:err:module:import_dll Library avfilter-9.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avdevice-60.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libssh.dll") not found
> 0024:err:module:import_dll Library libssh.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libwinpthread-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libsrt.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libp11-kit-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libidn2-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libunistring-5.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libtasn1-6.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libintl-8.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libgnutls-30.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library avformat-60.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avdevice-60.dll") not found
> 0024:err:module:import_dll Library avdevice-60.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\ffmpeg.exe") not found
> 0024:err:module:import_dll Library libglib-2.0-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libharfbuzz-0.dll") not found
> 0024:err:module:import_dll Library libharfbuzz-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 0024:err:module:import_dll Library libexpat-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libfontconfig-1.dll") not found
> 0024:err:module:import_dll Library libfontconfig-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libssh.dll") not found
> 0024:err:module:import_dll Library libssh.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libwinpthread-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libsrt.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libp11-kit-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libidn2-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libunistring-5.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libtasn1-6.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libintl-8.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libgnutls-30.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library avformat-60.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avfilter-9.dll") not found
> 0024:err:module:import_dll Library avfilter-9.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\ffmpeg.exe") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libssh.dll") not found
> 0024:err:module:import_dll Library libssh.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libwinpthread-1.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libcrypto-3.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libsrt.dll") not found
> 0024:err:module:import_dll Library libsrt.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library libp11-kit-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libidn2-0.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libunistring-5.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libtasn1-6.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libintl-8.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\libgnutls-30.dll") not found
> 0024:err:module:import_dll Library libgnutls-30.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\avformat-60.dll") not found
> 0024:err:module:import_dll Library avformat-60.dll (which is needed by L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\ffmpeg.exe") not found
> 0024:err:module:loader_init Importing dlls for L"Z:\\Users\\atemu\\Projects\\nixpkgs\\result-bin\\bin\\ffmpeg.exe" failed, status c0000135
> ```
> 
> Did you test this on actual Windows?

You can't just run wine directly, I needed to create an installer. When I created the installer, I needed to manually copy dll's, (no sym links) to the same location where the exe existed. I have run a qtmultimedia example application which has ffmpeg as a dependency. I ran this on actual windows.

File: pkgs/development/libraries/ffmpeg/generic.nix (hunk 2 above)
Comment (seanybaggins) [reply pkgs/development/libraries/ffmpeg/generic.nix]: Applied suggested change

Comment (jopejoe1) [reply pkgs/development/libraries/ffmpeg/generic.nix]: I thought networking was the reason it is avaible on windows and darwin.

eg. https://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/User/Network/

**Comment by Atemu** [comment]
Oookay windows. Nice sane operating system you got there.

Reminder to self: There's a reason you went away from Windows.

**Review by Atemu — APPROVED** [review]
I was able to repro the build failure and success, diff LGTM.

**Review by jopejoe1 — APPROVED** [review]

**Comment by seanybaggins** [comment]
> Oookay windows. Nice sane operating system you got there.
> 
> Reminder to self: There's a reason you went away from Windows.

Yeah not my favorite. My company needs to support both windows and linux for our Qt Applications. This is why this work even exists.

**Comment by Atemu** [comment]
And you're using Nix to build that?

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 3]
```diff
@@ -722,7 +722,7 @@ stdenv.mkDerivation (finalAttrs: {
     addOpenGLRunpath ${placeholder "lib"}/lib/libavutil.so
   ''
   # https://trac.ffmpeg.org/ticket/10809
-  + optionalString (versionAtLeast version "5.0" && withVulkan) ''
+  + optionalString (versionAtLeast version "5.0" && withVulkan && !stdenv.hostPlatform.isMinGW) ''
```
Comment (Atemu) [inline pkgs/development/libraries/ffmpeg/generic.nix:725]: Oh, this should probably be guarded behind isLinux instead as this .so patching is quite linux-specific.

**Comment by seanybaggins** [comment]
> And you're using Nix to build that?

I am using nix to build all the dependencies and create the .iss file. I need to manually run inno setup to do the final packaging. Working on seeing if I can add inno setup to nixpkgs

**Comment by Atemu** [comment]
Cool!

**Pull request merged**